### PR TITLE
FIX Two CMSField fixes (related to UploadField / FileBlock)

### LINKED ISSUE #452: Unable to upload files inline
Running the latest elemental `dev-master f7a4c73826` it is not possible to add a file to a file or banner block through the inline interface.

### PR DESCRIPTION
**Pull request by ScopeyNZ**
Fixes #452

- Fields that are also request handlers are now supported
- Fields that have submitted values different from saved values are now supported

### CODE REVIEW

File: src/Controllers/ElementalAreaController.php [hunk 1]
```diff
@@ -159,6 +161,19 @@ public function apiSaveForm(HTTPRequest $request)
         return HTTPResponse::create($body)->addHeader('Content-Type', 'application/json');
     }
 
+    public function formAction(HTTPRequest $request)
```
Comment (robbieaverill) [inline src/Controllers/ElementalAreaController.php]: Can we have a doc block explaining why this method exists please?

Comment (raissanorth) [reply src/Controllers/ElementalAreaController.php]: Thanks for adding the doc block!

**Comment by raissanorth** [comment]
I have pulled the changes and tried it out locally. So far the file only gets saved when the entire page is saved, but not when an individual element is saved via the button in its action menu. 😞

**Comment by NightJar** [comment]
This is still waiting for corresponding work in https://github.com/silverstripe/silverstripe-admin/pull/711
Upload field sort-of needs to work in order to have this be validated for merge :)

**Pull request merged**